third_party/move/move-model/src/well_known.rs

```diff
@@ -44,3 +44,9 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
 
 /// The well-known name of the first parameter of a method.
 pub const RECEIVER_PARAM_NAME: &str = "self";
+
+/// The well-known abort codes used by the compiler. These are marked
+/// by lowest 6 bytes of a sha256 of the string "Move 2 Abort Code",
+/// appended with two bytes for the error type.
+pub const WELL_KNOWN_ABORT_CODE_BASE: u64 = 0xD8CA26CBD9BE << 16;
+pub const INCOMPLETE_MATCH_ABORT_CODE: u64 = WELL_KNOWN_ABORT_CODE_BASE | 0x1;
```

Comment:
minor: We allow user to also directly abort with this code (using abort, or assert)? Can this cause any issues (by using the abort codes in the user space)? As far as I understand, arithmetic errors, for example, are always distinguishable as an outcome from user aborts; should we have the same here?